examples/xhr-proxy/xhrproxy.js

```diff
@@ -0,0 +1,55 @@
+/* This is an example how to monkey-patch XMLHttpRequest() for HTTP requests to a local service port.
+ * Cockpit pages cannot use the standard XMLHttpRequest() due to its strict Content-Security-Policy,
+ * so these need to be wrapped into a raw cockpit channel and be "really" done from the host where
+ * the session runs.
+ *
+ * Note that non-localhost connections are still forbidden by this page's Content-Security-Policy
+ * (which can be changed in the manifest), and browsers blocking Cross-Origin requests.
+ *
+ * Documentation links:
+ * - https://cockpit-project.org/guide/latest/cockpit-channels.html
+ * - doc/protocol.md
+ * - doc/urls.md
+ */
+
+const origXHROpen = XMLHttpRequest.prototype.open;
+XMLHttpRequest.prototype.open = function(method, url) {
```

Comment:
If this is called before cockpit.transport.wait() it'll fail, since csrf_token is not yet set. Obviously this example only calls this on click, so the race is unlikely. But if people are going to copy-pasta this example, that's probably worth accounting for.